// diflexmo-angular-design/esm2020/lib/design-system/modules/datepicker/datepicker.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { DatepickerComponent } from './datepicker.component';
import { InputModule } from '../input/input.module';
import { ButtonModule } from '../button/button.module';
import { IconModule } from '../icon/icon.module';
import * as i0 from "@angular/core";
export class DatepickerModule {
}
DatepickerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: DatepickerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DatepickerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: DatepickerModule, declarations: [DatepickerComponent], imports: [CommonModule, InputModule, ButtonModule, ReactiveFormsModule, IconModule], exports: [DatepickerComponent] });
DatepickerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: DatepickerModule, imports: [CommonModule, InputModule, ButtonModule, ReactiveFormsModule, IconModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: DatepickerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DatepickerComponent],
                    imports: [CommonModule, InputModule, ButtonModule, ReactiveFormsModule, IconModule],
                    exports: [DatepickerComponent],
                }]
        }] });
