// diflexmo-angular-design/esm2020/lib/design-system/modules/table/table.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AngularResizeEventModule } from 'angular-resize-event';
import { FormsModule } from '@angular/forms';
import { InfiniteScrollModule } from 'ngx-infinite-scroll';
import { TableComponent } from './components/table/table.component';
import { TableHeaderCellComponent } from './components/table-header-cell/table-header-cell.component';
import { IconModule } from '../icon/icon.module';
import { TableBodyCellComponent } from './components/table-body-cell/table-body-cell.component';
import { BaseCheckboxModule } from '../base-checkbox/base-checkbox.module';
import * as i0 from "@angular/core";
export class TableModule {
}
TableModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: TableModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TableModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: TableModule, declarations: [TableComponent, TableHeaderCellComponent, TableBodyCellComponent], imports: [CommonModule, IconModule, AngularResizeEventModule, BaseCheckboxModule, FormsModule, InfiniteScrollModule], exports: [TableComponent, TableHeaderCellComponent, TableBodyCellComponent] });
TableModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: TableModule, imports: [CommonModule, IconModule, AngularResizeEventModule, BaseCheckboxModule, FormsModule, InfiniteScrollModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: TableModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [TableComponent, TableHeaderCellComponent, TableBodyCellComponent],
                    imports: [CommonModule, IconModule, AngularResizeEventModule, BaseCheckboxModule, FormsModule, InfiniteScrollModule],
                    exports: [TableComponent, TableHeaderCellComponent, TableBodyCellComponent],
                }]
        }] });
