// diflexmo-angular-design/esm2020/lib/design-system/modules/base-checkbox/base-checkbox.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { RadioButtonComponent } from './components/radio-button/radio-button.component';
import { CheckboxComponent } from './components/checkbox/checkbox.component';
import { ToggleComponent } from './components/toggle/toggle.component';
import { BaseCheckboxComponent } from './components/base-checkbox.component';
import * as i0 from "@angular/core";
export class BaseCheckboxModule {
}
BaseCheckboxModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: BaseCheckboxModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BaseCheckboxModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: BaseCheckboxModule, declarations: [BaseCheckboxComponent, RadioButtonComponent, CheckboxComponent, ToggleComponent], imports: [CommonModule, FormsModule], exports: [RadioButtonComponent, CheckboxComponent, ToggleComponent] });
BaseCheckboxModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: BaseCheckboxModule, imports: [CommonModule, FormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: BaseCheckboxModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BaseCheckboxComponent, RadioButtonComponent, CheckboxComponent, ToggleComponent],
                    imports: [CommonModule, FormsModule],
                    exports: [RadioButtonComponent, CheckboxComponent, ToggleComponent],
                }]
        }] });
