// diflexmo-angular-design/esm2020/lib/design-system/modules/clipboard/clipboard.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TranslateModule } from '@ngx-translate/core';
import { ClipboardComponent } from './clipboard.component';
import { IconModule } from '../icon/icon.module';
import * as i0 from "@angular/core";
import * as i1 from "@ngx-translate/core";
export class ClipboardModule {
}
ClipboardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ClipboardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ClipboardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: ClipboardModule, declarations: [ClipboardComponent], imports: [CommonModule, IconModule, i1.TranslateModule], exports: [ClipboardComponent] });
ClipboardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ClipboardModule, imports: [CommonModule, IconModule, TranslateModule.forChild()] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ClipboardModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ClipboardComponent],
                    imports: [CommonModule, IconModule, TranslateModule.forChild()],
                    exports: [ClipboardComponent],
                }]
        }] });
