// diflexmo-angular-design/esm2020/lib/design-system/modules/clipboard/clipboard.component.mjs
import { Component, ElementRef, Input, ViewChild } from '@angular/core';
import { Tooltip } from 'bootstrap';
import * as i0 from "@angular/core";
import * as i1 from "../../services/clipboard.service";
import * as i2 from "@ngx-translate/core";
import * as i3 from "@angular/common";
import * as i4 from "../icon/components/icon/icon.component";
export class ClipboardComponent {
    constructor(clipboardSerice, translateService) {
        this.clipboardSerice = clipboardSerice;
        this.translateService = translateService;
        this.clip = '';
        this.alwaysVisible = false;
        this.isCopied = false;
        this.copyToClipboardText = 'Copy to clipboard';
        this.copiedToClipboardText = 'Copied!';
        this.copyToClipboardTextKey = 'DESIGN_SYSTEM.MODULES.CLIPBOARD.COPY_TEXT';
        this.copiedToClipboardTextKey = 'DESIGN_SYSTEM.MODULES.CLIPBOARD.COPIED_TEXT';
        this.tooltips = [];
    }
    ngOnInit() {
        this.translateService.get(this.copyToClipboardTextKey).subscribe((t) => {
            if (t !== this.copyToClipboardTextKey) {
                this.copyToClipboardText = t;
            }
        });
        this.translateService.get(this.copiedToClipboardTextKey).subscribe((t) => {
            if (t !== this.copiedToClipboardTextKey) {
                this.copiedToClipboardText = t;
            }
        });
    }
    ngAfterViewInit() {
        const copyTooltip = new Tooltip(this.copyIcon.nativeElement, { title: this.copyToClipboardText });
        const copiedTooltip = new Tooltip(this.copiedIcon.nativeElement, { title: this.copiedToClipboardText });
        this.tooltips.push(copyTooltip, copiedTooltip);
    }
    ngOnDestroy() {
        this.tooltips.forEach((t) => t.dispose());
    }
    copyToClipboard() {
        this.clipboardSerice.emitToClipboard(this.clip);
        this.isCopied = true;
        setTimeout(() => {
            this.isCopied = false;
        }, 1000);
    }
}
ClipboardComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ClipboardComponent, deps: [{ token: i1.ClipboardService }, { token: i2.TranslateService }], target: i0.ɵɵFactoryTarget.Component });
ClipboardComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.12", type: ClipboardComponent, selector: "dfm-clipboard", inputs: { clip: "clip", alwaysVisible: "alwaysVisible" }, viewQueries: [{ propertyName: "copyIcon", first: true, predicate: ["copyIcon"], descendants: true, read: ElementRef }, { propertyName: "copiedIcon", first: true, predicate: ["copiedIcon"], descendants: true, read: ElementRef }], ngImport: i0, template: `
    <div class="d-flex dfm-gap-8 show-hidden">
      <div><ng-content></ng-content></div>
      <dfm-icon
        #copyIcon
        (click)="copyToClipboard(); $event.stopPropagation()"
        class="dfm-clipboard pointer icon-15 align-self-center"
        [ngClass]="{ hide: !alwaysVisible, 'd-none': isCopied }"
        name="copy-06"
      >
      </dfm-icon>
      <dfm-icon
        #copiedIcon
        (click)="$event.stopPropagation()"
        name="check"
        class="dfm-clipboard-copied pointer icon-15 align-self-center"
        [ngClass]="{ hide: !alwaysVisible, 'd-none': !isCopied }"
      ></dfm-icon>
    </div>
  `, isInline: true, styles: [".dfm-clipboard{color:var(--dfm-clipboard-color)}.dfm-clipboard:hover{color:var(--dfm-clipboard-color-hover)}.dfm-clipboard.hidden{display:none}.dfm-clipboard-copied{color:var(--dfm-clipboard-color-copied)}.dfm-clipboard-copied.hidden{display:none}\n"], dependencies: [{ kind: "directive", type: i3.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "component", type: i4.IconComponent, selector: "dfm-icon", inputs: ["name"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ClipboardComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dfm-clipboard', template: `
    <div class="d-flex dfm-gap-8 show-hidden">
      <div><ng-content></ng-content></div>
      <dfm-icon
        #copyIcon
        (click)="copyToClipboard(); $event.stopPropagation()"
        class="dfm-clipboard pointer icon-15 align-self-center"
        [ngClass]="{ hide: !alwaysVisible, 'd-none': isCopied }"
        name="copy-06"
      >
      </dfm-icon>
      <dfm-icon
        #copiedIcon
        (click)="$event.stopPropagation()"
        name="check"
        class="dfm-clipboard-copied pointer icon-15 align-self-center"
        [ngClass]="{ hide: !alwaysVisible, 'd-none': !isCopied }"
      ></dfm-icon>
    </div>
  `, styles: [".dfm-clipboard{color:var(--dfm-clipboard-color)}.dfm-clipboard:hover{color:var(--dfm-clipboard-color-hover)}.dfm-clipboard.hidden{display:none}.dfm-clipboard-copied{color:var(--dfm-clipboard-color-copied)}.dfm-clipboard-copied.hidden{display:none}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.ClipboardService }, { type: i2.TranslateService }]; }, propDecorators: { clip: [{
                type: Input
            }], alwaysVisible: [{
                type: Input
            }], copyIcon: [{
                type: ViewChild,
                args: ['copyIcon', { read: ElementRef }]
            }], copiedIcon: [{
                type: ViewChild,
                args: ['copiedIcon', { read: ElementRef }]
            }] } });
