// diflexmo-angular-design/esm2020/lib/design-system/services/clipboard.service.mjs
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import { map, debounceTime } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@ngx-translate/core";
export class ClipboardService {
    constructor(translationService) {
        this.translationService = translationService;
        this.copySubject = new Subject();
        this.copySubject
            .pipe(debounceTime(300), map((x) => this.copyToClipboard(x)))
            .subscribe();
    }
    emitToClipboard(value) {
        this.copySubject.next(value);
    }
    copyToClipboard(value) {
        // https://stackoverflow.com/questions/49102724/angular-5-copy-to-clipboard
        const selBox = document.createElement('textarea');
        selBox.style.position = 'fixed';
        selBox.style.left = '0';
        selBox.style.top = '0';
        selBox.style.opacity = '0';
        selBox.value = value;
        document.body.appendChild(selBox);
        selBox.focus();
        selBox.select();
        document.execCommand('copy');
        document.body.removeChild(selBox);
    }
}
ClipboardService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ClipboardService, deps: [{ token: i1.TranslateService }], target: i0.ɵɵFactoryTarget.Injectable });
ClipboardService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ClipboardService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ClipboardService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return [{ type: i1.TranslateService }]; } });
