// diflexmo-angular-design/esm2020/lib/design-system/modules/base-checkbox/components/base-checkbox.component.mjs
import { Component, Input } from '@angular/core';
import { BaseControlValueAccessor } from '../../base-control-value-accessor';
import * as i0 from "@angular/core";
export class BaseCheckboxComponent extends BaseControlValueAccessor {
    constructor() {
        super();
        this.label = '';
        this.hint = '';
        this.size = 'md';
        this.id = Math.random().toString(36).slice(2, 7);
        this.value = false;
    }
}
BaseCheckboxComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: BaseCheckboxComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BaseCheckboxComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.12", type: BaseCheckboxComponent, selector: "dfm-base-checkbox", inputs: { label: "label", hint: "hint", size: "size" }, usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: BaseCheckboxComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dfm-base-checkbox', template: '' }]
        }], ctorParameters: function () { return []; }, propDecorators: { label: [{
                type: Input
            }], hint: [{
                type: Input
            }], size: [{
                type: Input
            }] } });
