// diflexmo-angular-design/esm2020/lib/design-system/modules/error/error.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { ErrorComponent } from './error.component';
import * as i0 from "@angular/core";
export class ErrorModule {
}
ErrorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ErrorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ErrorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: ErrorModule, declarations: [ErrorComponent], imports: [CommonModule, FormsModule], exports: [ErrorComponent] });
ErrorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ErrorModule, imports: [CommonModule, FormsModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: ErrorModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ErrorComponent],
                    imports: [CommonModule, FormsModule],
                    exports: [ErrorComponent],
                }]
        }] });
