// diflexmo-angular-design/esm2020/lib/design-system/modules/card/card.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../icon/components/icon/icon.component";
export class CardComponent {
    constructor() {
        this.title = '';
        this.showHeader = true;
        this.collapseBody = false;
        this.isDraggable = true;
    }
}
CardComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: CardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CardComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.12", type: CardComponent, selector: "dfm-card", inputs: { title: "title", showHeader: "showHeader", collapseBody: "collapseBody", iconName: "iconName", isDraggable: "isDraggable" }, ngImport: i0, template: `
    <div class="dfm-card">
      <div class="dfm-card-header" *ngIf="showHeader">
        <div class="dfm-card-title d-flex dfm-gap-8">
          <div class="dfm-card-icon" [ngClass]="{'drag-handler pointer': isDraggable}">
            <ng-container *ngIf="isDraggable">
              <dfm-icon name="drag"></dfm-icon>
            </ng-container>
            <ng-container *ngIf="iconName && !isDraggable">
              <dfm-icon [name]="iconName"></dfm-icon>
            </ng-container>
          </div>
          <ng-container *ngIf="title; else contentTitle">
            {{ title }}
          </ng-container>
        </div>
        <div class="dfm-card-toolbar">
          <ng-content select="[toolbar]"></ng-content>
        </div>
      </div>

      <div *ngIf="!collapseBody" class="dfm-card-body-no-gutter">
        <ng-content></ng-content>
      </div>
    </div>

    <ng-template #contentTitle>
      <ng-content select="[title]"></ng-content>
    </ng-template>
  `, isInline: true, styles: [".dfm-card{background-color:var(--dfm-card-bg);box-shadow:0 1px 2px #1018280f;border-radius:20px;width:100%;height:100%}.dfm-card>.dfm-card-header{display:flex;justify-content:space-between;align-items:stretch;flex-wrap:wrap;padding:16px;margin-bottom:0;background-color:transparent;position:relative}.dfm-card-header>.dfm-card-title{align-items:center;font-family:Montserrat;font-style:normal;font-weight:500;font-size:16px;line-height:24px;color:var(--dfm-card-title-color)}.dfm-card-title>.dfm-card-icon{color:var(--dfm-card-icon-color)}.dfm-card-header>.dfm-card-toolbar{display:flex;flex-direction:row;flex-wrap:nowrap;gap:16px;align-items:center;flex-wrap:wrap;color:var(--dfm-card-toolbar-color)}.dfm-card>.dfm-card-body{padding-left:16px;padding-bottom:35px;padding-right:16px;word-wrap:break-word}.dfm-card>.dfm-card-body-no-gutter{word-wrap:break-word;height:calc(100% - 56px);border-bottom-left-radius:20px;border-bottom-right-radius:20px}\n"], dependencies: [{ kind: "directive", type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: i2.IconComponent, selector: "dfm-icon", inputs: ["name"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: CardComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dfm-card', template: `
    <div class="dfm-card">
      <div class="dfm-card-header" *ngIf="showHeader">
        <div class="dfm-card-title d-flex dfm-gap-8">
          <div class="dfm-card-icon" [ngClass]="{'drag-handler pointer': isDraggable}">
            <ng-container *ngIf="isDraggable">
              <dfm-icon name="drag"></dfm-icon>
            </ng-container>
            <ng-container *ngIf="iconName && !isDraggable">
              <dfm-icon [name]="iconName"></dfm-icon>
            </ng-container>
          </div>
          <ng-container *ngIf="title; else contentTitle">
            {{ title }}
          </ng-container>
        </div>
        <div class="dfm-card-toolbar">
          <ng-content select="[toolbar]"></ng-content>
        </div>
      </div>

      <div *ngIf="!collapseBody" class="dfm-card-body-no-gutter">
        <ng-content></ng-content>
      </div>
    </div>

    <ng-template #contentTitle>
      <ng-content select="[title]"></ng-content>
    </ng-template>
  `, styles: [".dfm-card{background-color:var(--dfm-card-bg);box-shadow:0 1px 2px #1018280f;border-radius:20px;width:100%;height:100%}.dfm-card>.dfm-card-header{display:flex;justify-content:space-between;align-items:stretch;flex-wrap:wrap;padding:16px;margin-bottom:0;background-color:transparent;position:relative}.dfm-card-header>.dfm-card-title{align-items:center;font-family:Montserrat;font-style:normal;font-weight:500;font-size:16px;line-height:24px;color:var(--dfm-card-title-color)}.dfm-card-title>.dfm-card-icon{color:var(--dfm-card-icon-color)}.dfm-card-header>.dfm-card-toolbar{display:flex;flex-direction:row;flex-wrap:nowrap;gap:16px;align-items:center;flex-wrap:wrap;color:var(--dfm-card-toolbar-color)}.dfm-card>.dfm-card-body{padding-left:16px;padding-bottom:35px;padding-right:16px;word-wrap:break-word}.dfm-card>.dfm-card-body-no-gutter{word-wrap:break-word;height:calc(100% - 56px);border-bottom-left-radius:20px;border-bottom-right-radius:20px}\n"] }]
        }], propDecorators: { title: [{
                type: Input
            }], showHeader: [{
                type: Input
            }], collapseBody: [{
                type: Input
            }], iconName: [{
                type: Input
            }], isDraggable: [{
                type: Input
            }] } });
