// diflexmo-angular-design/esm2020/lib/design-system/modules/input/input.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { InputComponent } from './input.component';
import { IconModule } from '../icon/icon.module';
import { ErrorModule } from '../error/error.module';
import * as i0 from "@angular/core";
export class InputModule {
}
InputModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: InputModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: InputModule, declarations: [InputComponent], imports: [CommonModule, FormsModule, ReactiveFormsModule, IconModule, ErrorModule], exports: [InputComponent] });
InputModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: InputModule, imports: [CommonModule, FormsModule, ReactiveFormsModule, IconModule, ErrorModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: InputModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [InputComponent],
                    imports: [CommonModule, FormsModule, ReactiveFormsModule, IconModule, ErrorModule],
                    exports: [InputComponent],
                }]
        }] });
