// diflexmo-angular-design/esm2020/lib/design-system/modules/notification/notification.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CardModule } from '../card/card.module';
import { NotificationComponent } from './components/notification/notification.component';
import { IconModule } from '../icon/icon.module';
import * as i0 from "@angular/core";
export class NotificationModule {
}
NotificationModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: NotificationModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NotificationModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: NotificationModule, declarations: [NotificationComponent], imports: [CommonModule, CardModule, IconModule], exports: [NotificationComponent] });
NotificationModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: NotificationModule, imports: [CommonModule, CardModule, IconModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: NotificationModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NotificationComponent],
                    exports: [NotificationComponent],
                    imports: [CommonModule, CardModule, IconModule],
                }]
        }] });
