// diflexmo-angular-design/esm2020/lib/design-system/modules/input-dropdown/input-dropdown.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { IconModule } from '../icon/icon.module';
import { InputModule } from '../input/input.module';
import { TagModule } from '../tag/tag.module';
import { InputDropdownComponent } from './input-dropdown.component';
import { ErrorModule } from '../error/error.module';
import { DirectivesModule } from '../../directives/directives.module';
import * as i0 from "@angular/core";
export class InputDropdownModule {
}
InputDropdownModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: InputDropdownModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InputDropdownModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: InputDropdownModule, declarations: [InputDropdownComponent], imports: [CommonModule, FormsModule, IconModule, InputModule, TagModule, ErrorModule, DirectivesModule], exports: [InputDropdownComponent] });
InputDropdownModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: InputDropdownModule, imports: [CommonModule, FormsModule, IconModule, InputModule, TagModule, ErrorModule, DirectivesModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: InputDropdownModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [InputDropdownComponent],
                    imports: [CommonModule, FormsModule, IconModule, InputModule, TagModule, ErrorModule, DirectivesModule],
                    exports: [InputDropdownComponent],
                }]
        }] });
