// diflexmo-angular-design/esm2020/lib/design-system/modules/text-area/text-area.component.mjs
import { Component, Input, Optional } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@angular/common";
import * as i3 from "../error/error.component";
export class TextAreaComponent {
    constructor(control) {
        this.control = control;
        this.label = '';
        this.placeholder = '';
        this.hint = '';
        this.disabled = false;
        this.value = '';
        // eslint-disable-next-line class-methods-use-this
        this.onChange = () => { };
        // eslint-disable-next-line class-methods-use-this
        this.onTouched = () => { };
        if (this.control != null) {
            this.control.valueAccessor = this;
        }
    }
    writeValue(obj) {
        this.value = obj;
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
}
TextAreaComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: TextAreaComponent, deps: [{ token: i1.NgControl, optional: true }], target: i0.ɵɵFactoryTarget.Component });
TextAreaComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.12", type: TextAreaComponent, selector: "dfm-text-area", inputs: { label: "label", placeholder: "placeholder", hint: "hint", disabled: "disabled" }, ngImport: i0, template: "<div class=\"form-group\">\n  <label *ngIf=\"label\" class=\"form-label\">{{ label }}</label>\n  <textarea\n    class=\"form-control dfm-text-area\"\n    rows=\"3\"\n    [ngClass]=\"{ 'hint-control': hint, 'is-invalid': control && control.touched && !control.valid }\"\n    [ngModel]=\"value\"\n    (ngModelChange)=\"onChange($event)\"\n    [disabled]=\"disabled\"\n    [placeholder]=\"placeholder\"\n    (blur)=\"onTouched($event)\"\n  >\n  </textarea>\n  <span *ngIf=\"hint\" class=\"dfm-input-hint\">{{ hint }}</span>\n  <dfm-error [control]=\"control\"></dfm-error>\n</div>\n", styles: [".dfm-text-area{font-family:Montserrat;font-style:normal;font-size:16px;line-height:24px}.dfm-text-area:focus{border:1px solid var(--dfm-text-area-focus-border-color);box-shadow:0 1px 2px #1018280d,0 0 0 1px var(--dfm-text-area-focus-shadow)}\n"], dependencies: [{ kind: "directive", type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i1.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }, { kind: "component", type: i3.ErrorComponent, selector: "dfm-error", inputs: ["control"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: TextAreaComponent, decorators: [{
            type: Component,
            args: [{ selector: 'dfm-text-area', template: "<div class=\"form-group\">\n  <label *ngIf=\"label\" class=\"form-label\">{{ label }}</label>\n  <textarea\n    class=\"form-control dfm-text-area\"\n    rows=\"3\"\n    [ngClass]=\"{ 'hint-control': hint, 'is-invalid': control && control.touched && !control.valid }\"\n    [ngModel]=\"value\"\n    (ngModelChange)=\"onChange($event)\"\n    [disabled]=\"disabled\"\n    [placeholder]=\"placeholder\"\n    (blur)=\"onTouched($event)\"\n  >\n  </textarea>\n  <span *ngIf=\"hint\" class=\"dfm-input-hint\">{{ hint }}</span>\n  <dfm-error [control]=\"control\"></dfm-error>\n</div>\n", styles: [".dfm-text-area{font-family:Montserrat;font-style:normal;font-size:16px;line-height:24px}.dfm-text-area:focus{border:1px solid var(--dfm-text-area-focus-border-color);box-shadow:0 1px 2px #1018280d,0 0 0 1px var(--dfm-text-area-focus-shadow)}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.NgControl, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { label: [{
                type: Input
            }], placeholder: [{
                type: Input
            }], hint: [{
                type: Input
            }], disabled: [{
                type: Input
            }] } });
