// diflexmo-angular-design/esm2020/lib/design-system/modules/text-area/text-area.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { TextAreaComponent } from './text-area.component';
import { ErrorModule } from '../error/error.module';
import * as i0 from "@angular/core";
export class TextAreaModule {
}
TextAreaModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: TextAreaModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TextAreaModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: TextAreaModule, declarations: [TextAreaComponent], imports: [CommonModule, FormsModule, ReactiveFormsModule, ErrorModule], exports: [TextAreaComponent] });
TextAreaModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: TextAreaModule, imports: [CommonModule, FormsModule, ReactiveFormsModule, ErrorModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: TextAreaModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [TextAreaComponent],
                    imports: [CommonModule, FormsModule, ReactiveFormsModule, ErrorModule],
                    exports: [TextAreaComponent],
                }]
        }] });
