// diflexmo-angular-design/esm2020/lib/design-system/modules/navigation-bar/navigation-bar.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { NavigationBarComponent } from './components/navigation-bar/navigation-bar.component';
import { NavigationItemComponent } from './components/navigation-item/navigation-item.component';
import { IconModule } from '../icon/icon.module';
import { NavigationItemTenantComponent } from './components/navigation-item-tenant/navigation-item-tenant.component';
import * as i0 from "@angular/core";
export class NavigationBarModule {
}
NavigationBarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: NavigationBarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NavigationBarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: NavigationBarModule, declarations: [NavigationBarComponent, NavigationItemComponent, NavigationItemTenantComponent], imports: [CommonModule, IconModule, RouterModule], exports: [NavigationBarComponent, NavigationItemComponent, NavigationItemTenantComponent] });
NavigationBarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: NavigationBarModule, imports: [CommonModule, IconModule, RouterModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: NavigationBarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NavigationBarComponent, NavigationItemComponent, NavigationItemTenantComponent],
                    imports: [CommonModule, IconModule, RouterModule],
                    exports: [NavigationBarComponent, NavigationItemComponent, NavigationItemTenantComponent],
                }]
        }] });
