// diflexmo-angular-design/esm2020/lib/design-system/modules/badge/badge.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IconModule } from '../icon/icon.module';
import { BadgeComponent } from './components/badge/badge.component';
import { BadgeIconComponent } from './components/badge-icon/badge-icon.component';
import * as i0 from "@angular/core";
export class BadgeModule {
}
BadgeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: BadgeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BadgeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.12", ngImport: i0, type: BadgeModule, declarations: [BadgeIconComponent, BadgeComponent], imports: [CommonModule, IconModule], exports: [BadgeIconComponent, BadgeComponent] });
BadgeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: BadgeModule, imports: [CommonModule, IconModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.12", ngImport: i0, type: BadgeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BadgeIconComponent, BadgeComponent],
                    imports: [CommonModule, IconModule],
                    exports: [BadgeIconComponent, BadgeComponent],
                }]
        }] });
